AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationStrategies/MagneticAllocation/MagneticAllocation.py

```diff
@@ -77,29 +86,34 @@ def allocate(
             self.center[1] - first_word.text_size[1] / 2,
         )
 
-        # register
-        self.rects_outermost.add(
-            Rect(
-                first_word_position,
-                (
-                    first_word_position[0] + first_word.text_size[0],
-                    first_word_position[1] + first_word.text_size[1],
-                ),
-            )
+        # register the first word
+        rect_adding = Rect(
+            first_word_position,
+            (
+                first_word_position[0] + first_word.text_size[0],
+                first_word_position[1] + first_word.text_size[1],
+            ),
         )
+        self.rects.add(rect_adding)
         output.add(first_word.text, first_word.font_size, first_word_position)
 
+        # verbose for iteration
+        if self.verbosity == "debug":
+            print("MagneticAllocation: Start iteration...")
+            iterator = tqdm(self.words[1:])
+        else:
+            iterator = self.words[1:]
+
         # from second word
-        for word in self.words[1:]:
-            (
-                magnet_outer_frontier,
-                self.rects_outermost,
-            ) = get_magnet_outer_frontier(
-                self.rects_outermost,
+        for word in iterator:
+            magnet_outer_frontier = get_magnet_outer_frontier(
```

Comment:
Please explain what get_magnet_outer_frontier is called for and what magnet_outer_frontier is.

Reply:
Got it,
Check again please